src/utils.jl

```diff
@@ -105,3 +105,92 @@ function uncertainty_components(x::Measurement{T}) where {T<:AbstractFloat}
     end
     return out
 end
+
+"""
+    Measurements.covariance_matrix(n::AbstractVector{<:Measurement})
+
+Returns the covariance matrix of a vector of correlated `Measurement`s.
+"""
+function covariance_matrix(n::AbstractVector{<:Measurement})
+    S = length(n)
+    covariance_matrix = zeros(Float64, (S, S))
+
+    for i = 1:S, j = 1:i
+        covariance_matrix[i, j] = sum(keys(n[i].der) ∩ keys(n[j].der); init = 0.0) do var
```

Comment:
The current build failure happens on this line, where the `init` keyword was only added on version 1.6.
A possible fix compatible with version 1.0 would be
```suggestion
        overlap = keys(n[i].der) ∩ keys(n[j].der)
        covariance_matrix[i, j] = isempty(overlap) ? 0.0 : sum(overlap) do var
```
I am not sure if that would even impact the performance, and, unless you want to calculate the covariance matrix of thousands of measurements, it would not matter.

I will have a look over the next few days if there are major roadblocks to a 1.0 compatibility, and what version would be an acceptable compromise.

Reply:
v1.6 is currently the LTS, it'd be ok to require that version :+1:

Reply:
That's great, I would have suggested the same thing. Then I will check if everything runs on 1.6 and fix all remaining issues.

Reply:
Actually, I just tested the code with 1.6 and 1.0:
- On 1.6, the rest of the code (meaning all the tests) pass in the current state of the PR.
- On 1.0, all tests pass if I apply my suggested change from above.
I would proceed by applying my change and keeping the 1.0 version compatibility if you agree.
If you think that it would anyway make sense to update the version requirements, we can also do that. Then we could keep it a bit more readable and elegant, and maybe open up some improved syntax and newer functions for the rest of the codebase.

Reply:
So far I kept the approach of supporting 1.0 as long as it isn't too much extra effort.  It looks like this wouldn't require _too much_ effort, but I think at some point is fine stopping supporting 1.0, and remove a bunch of `if VERSION` or `if isdefined` that are around the package and simplify the code.